src/components/Card.js

```diff
@@ -1,41 +1,82 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { _id, name, link, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleConfirmDelete,
+    handleCardLike,
+    handleCardDisLike
+  ) {
+    this._id = _id;
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleConfirmDelete = handleConfirmDelete;
+    this._handleCardLike = handleCardLike;
+    this._isLiked = isLiked;
+    this._handleCardDisLike = handleCardDisLike;
   }
 
   _setEventListeners() {
     //".card__like-button"
     this._element
       .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeButton();
+      .addEventListener("click", (e) => {
+        e.preventDefault();
+
+        if (this._isLiked) {
+          this._handleCardDisLike(this);
+          this._likeButton.classList.remove("card__like-button_active");
+        } else {
+          this._handleCardLike(this);
+          this._likeButton.classList.add("card__like-button_active");
+        }
       });
 
     //".card__delete-button"
     this._element
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteButton();
+        this._handleConfirmDelete(this);
       });
 
     this._cardImage.addEventListener("click", () => {
-     this._handleImageClick({ name: this._name, link: this._link });
+      this._handleImageClick({ name: this._name, link: this._link });
+    });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleConfirmDelete(this);
     });
   }
 
   _handleLikeButton() {
-    this._element
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+    //this._element
+    //.querySelector(".card__like-button")
```

Comment:
You've defined `this._likeButton` for the like button in line 91. Please, use it everywhere

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.